src/components/common/seo.tsx

```diff
@@ -1,24 +1,38 @@
 import React from "react";
-import PropTypes from "prop-types";
 import { Helmet } from "react-helmet";
 import { useStaticQuery, graphql } from "gatsby";
 
-function SEO({ description, lang, meta, title }) {
-  const { site } = useStaticQuery(
+const getSocialImageSrc = (image: { src; alt }, site) => {
+  const relativeImagePath = image.src;
```

Comment:
I think it would be better to have the parameters to this function be the `imageSrc` and `siteHostName` to avoid having to pass the full `image` and `site` objects in. It'd make this more easily testable too. 

Speaking of which, are there any associated snapshot or unit tests that should be updated or added as a result of the changes in this PR?

Reply:
There are no unit tests looking at metadata at all. I can add these.

Reply:
Added some spec for the helpers, and split out a PureSeo version of the Seo component – although the snapshot seems to have some issues rendering `Helmet` from `react-helmet`, it's at least some insurance now, especially in combination with the helper checker.